subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/AbstractDependenciesMetadataAdapter.java

```diff
@@ -32,23 +32,27 @@
 import java.util.List;
 import java.util.Map;
 
-public class DependenciesMetadataAdapter extends AbstractList<DependencyMetadata> implements DependenciesMetadata {
+abstract public class AbstractDependenciesMetadataAdapter<T extends DependencyMetadata> extends AbstractList<T> implements DependenciesMetadata<T> {
```

Comment:
I'd be keen on renaming `DependenciesMetadata` to something else. Doing the review it's often hard to make the difference between `DependencyMetadata` and `DependenciesMetadata` in a look. Maybe something like `DependencyMetadataContainer` would be more consistent with what we have in other places of the code.

Reply:
Also weird ordering of modifiers :)

Reply:
Thanks. Fixed.